picoquic/frames.c

```diff
@@ -1808,8 +1821,6 @@ int picoquic_prepare_misc_frame(picoquic_cnx_t* cnx, uint8_t* bytes,
         uint8_t* frame = ((uint8_t*)misc_frame) + sizeof(picoquic_misc_frame_header_t);
         memcpy(bytes, frame, misc_frame->length);
         *consumed = misc_frame->length;
-        cnx->first_misc_frame = misc_frame->next_misc_frame;
-        free(misc_frame);
     }
```

Comment:
Removing the free(frame) statement will create a memory leak if we keep the definition of the original API. Do we really need this?